The hypotheses for the Chi-Square test are:

Null Hypothesis (H0): There is no association between the type of smart home device purchased and the customer satisfaction level.

Alternative Hypothesis (H1): There is an association between the type of smart home device purchased and the customer satisfaction level.

In [1]:
import numpy as np
import scipy.stats as stats

# Contingency table
data = np.array([[50, 70],
                 [80, 100],
                 [60, 90],
                 [30, 50],
                 [20, 50]])

#Chi-Square test
chi2_stat, p_val, dof, expected = stats.chi2_contingency(data)

print('Chi-Square Statistic:', chi2_stat)
print('P-Value:', p_val)
print('Degrees of Freedom:', dof)
print('Expected Frequencies:')
print(expected)

Chi-Square Statistic: 5.638227513227513
P-Value: 0.22784371130697179
Degrees of Freedom: 4
Expected Frequencies:
[[ 48.  72.]
 [ 72. 108.]
 [ 60.  90.]
 [ 32.  48.]
 [ 28.  42.]]


In [2]:
#Determining the Critical Value
alpha = 0.05
critical_value = stats.chi2.ppf(1 - alpha, dof)
print('Critical Value:', critical_value)

Critical Value: 9.487729036781154


In [3]:
#Decision Making
if chi2_stat > critical_value:
    print('Reject the null hypothesis (H0): There is an association between device type and customer satisfaction.')
else:
    print('Fail to reject the null hypothesis (H0): There is no association between device type and customer satisfaction.')

Fail to reject the null hypothesis (H0): There is no association between device type and customer satisfaction.


Conclusion:
After conducting the Chi-Square test for independence to examine the relationship between
the type of smart home device (Smart Thermostats vs. Smart Lights) and customer satisfaction levels, the results are as follows:

Chi-Square Statistic: 5.64
Degrees of Freedom: 4
Critical Value: 9.49
P-Value: 0.23

Since the Chi-Square statistic (5.64) is less than the critical value (9.49)
and the P-value (0.23) is greater than the significance level of 0.05, we fail to reject the null hypothesis (H0).